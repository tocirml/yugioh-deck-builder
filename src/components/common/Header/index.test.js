import React from 'react';
import { Header } from './index';
import { NavLink, MemoryRouter } from 'react-router-dom';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';

import { shallow } from 'enzyme';

describe('rendering', () => {
  it('renders without crashing', () => {
    shallow(<Header />);
  });

  it('renders 1 NavLink', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(NavLink).length).toBe(1);
  });
});

describe('mobile navbar', () => {
  //I have no idea how this is working
  it('toggles mobile menu correctly', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    expect(screen.getByTestId('navbar')).not.toHaveClass('mobile');
    expect(screen.getByTestId('login')).not.toHaveClass('mobile');

    userEvent.click(screen.getByTestId('anvorgesa'));

    expect(screen.getByTestId('navbar')).toHaveClass('mobile');
    expect(screen.getByTestId('login')).toHaveClass('mobile');

    userEvent.click(screen.getByTestId('anvorgesa'));

    expect(screen.getByTestId('navbar')).not.toHaveClass('mobile');
    expect(screen.getByTestId('login')).not.toHaveClass('mobile');
  });
});

describe('correct links', () => {
  render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  const homeLink = screen.getByText('DECK BUILDER').closest('a');

  it('Home has correct href', () => {
    expect(homeLink).toHaveAttribute('href', '/');
  });
});

describe('snapshots', () => {
  it('Header snapshot', () => {
    const tree = shallow(<Header />);
    expect(tree).toMatchSnapshot();
  });
});
