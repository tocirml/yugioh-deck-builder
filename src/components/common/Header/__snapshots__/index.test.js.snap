// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Header snapshot 1`] = `
<header
  className="header"
>
  <NavLink
    to="/"
  >
    <div
      className="logo"
    >
      <img
        alt="yugioh logo"
        src="test-file-stub"
      />
      <div
        className="logo-text"
      >
        DECK BUILDER
      </div>
    </div>
  </NavLink>
  <Nav
    mobile={false}
  />
  <Login
    mobile={false}
  />
  <Anvorgesa
    toggleHandler={[Function]}
  />
</header>
`;
